src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -274,6 +287,24 @@ public String whoAmI() throws IOException, InterruptedException {
 
     }
 
+    /**
+     * Checks if this {@link DockerClient} instance is running inside a container and returns the id of the container
+     * if so.
+     *
+     * @return a {@link Optional<String>} containing the <strong>container id</strong>, or <strong>absent</strong> if
+     * it isn't containerized.
+     */
+    public Optional<String> getContainerIdIfContainerized() throws IOException, InterruptedException {
+        ByteArrayOutputStream cgroup = new ByteArrayOutputStream();
+        launcher.launch().cmds("cat", "/proc/self/cgroup").quiet(true).stdout(cgroup).join();
```

Comment:
So if running on a slave, this will get `/proc/self/cgroup` from the slave, not the master. Which is I think what you want. But have you tested it?

BTW rather than using `Launcher` to `cat`, would be easier to just

```java
String cgroup = node.createPath("/proc/self/cgroup").readToString();
```